kclvm/parser/src/parser/stmt.rs

```diff
@@ -1420,7 +1425,8 @@ impl<'a> Parser<'_> {
                         .unwrap();
                     formatted_value.format_spec = Some(format_spec);
                 } else {
-                    panic!("invalid joined string spec");
+                    this.sess
+                        .struct_span_error("invalid joined string spec", parser.token.span);
```

Comment:
More friendly error messages are required. For example, which tokens should be expected and what a correct string spec looks like.